org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleBuildSupport.java

```diff
@@ -106,7 +107,17 @@ public void update(IProject project, boolean force, IProgressMonitor monitor) th
 					|| (settingsFile.exists() && JavaLanguageServerPlugin.getDigestStore().updateDigest(settingsFile.toPath()))
 					|| (buildKtsFile.exists() && JavaLanguageServerPlugin.getDigestStore().updateDigest(buildKtsFile.toPath()))
 					|| (settingsKtsFile.exists() && JavaLanguageServerPlugin.getDigestStore().updateDigest(settingsKtsFile.toPath()));
-			if (isRoot || shouldUpdate) {
+			// https://github.com/redhat-developer/vscode-java/issues/3893
+			shouldUpdate = isRoot || shouldUpdate;
+			if (!shouldUpdate) {
+				if (force && gradleBuild instanceof DefaultGradleBuild defaultGradleBuild) {
+					org.eclipse.buildship.core.internal.configuration.BuildConfiguration gradleConfig = defaultGradleBuild.getBuildConfig();
+					if (!gradleConfig.isAutoSync()) {
+						shouldUpdate = true;
+					}
+				}
+			}
+			if (shouldUpdate) {
```

Comment:
Just to confirm. This part is meant to fix the `interactive` case right ? Seems to only get triggered from the client-side interaction through `java/projectConfigurationUpdate`. I guess it might also get triggered if auto sync is off and the build configuration setting is set to `automatic` ?